### PR TITLE
CIRC-262 Implement sending request-related notices (user-initiated events)

### PR DESCRIPTION
**Pull request by dmytrotur**
## Purpose
Resolves: CIRC-262
 Implement sending request-related notices (user-initiated events)
## Links
https://issues.folio.org/browse/CIRC-262

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java [hunk 1]
```diff
@@ -2,37 +2,66 @@
...
   }
 
-  public static JsonObject createNoticeContextFromLoan(Loan loan) {
-    return createNoticeContextFromLoan(loan, DateTimeZone.UTC);
+  public static JsonObject createLoanNoticeContext(Loan loan) {
+    return createLoanNoticeContext(loan, DateTimeZone.UTC);
   }
 
-  public static JsonObject createNoticeContextFromLoan(Loan loan, DateTimeZone timeZone) {
+  public static JsonObject createLoanNoticeContext(Loan loan, DateTimeZone timeZone) {
```
Comment (mreno-EBSCO) [inline src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java]: Can't we do something like this to reduce code duplication:
```suggestion
  public static JsonObject createLoanNoticeContext(Loan loan, DateTimeZone timeZone) {
    return createNoticeContextFromItem(loan.getItem(), loan.getUser())
        .put("dueDate", loan.getDueDate().withZone(timeZone).toString());
  }
```

File: src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java [hunk 2]
```diff
@@ -2,37 +2,66 @@
...
+    return new JsonObject()
+      .put(PATRON, patron)
+      .put(ITEM, itemContext)
+      .put("dueDate", loan.getDueDate().withZone(timeZone).toString());
+  }
+
+  public static JsonObject createRequestNoticeContext(Request request) {
+    JsonObject patron = createPatron(request.getRequester());
+    JsonObject itemContext = createItemContext(request.getItem());
+
```
Comment (mreno-EBSCO) [inline src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java]: Can't we do something like this to reduce code duplication:
```suggestion
  public static JsonObject createRequestNoticeContext(Request request) {
    return createNoticeContextFromItem(request.getItem(), request.getRequester());
  }
```

File: src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java (hunk 1 above)
Comment (dmytrotur) [reply src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java]: Done

File: src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java (hunk 2 above)
Comment (dmytrotur) [reply src/main/java/org/folio/circulation/domain/notice/NoticeContextUtil.java]: Done

**Pull request merged**